ilastik/applets/dataSelection/opDataSelection.py

```diff
@@ -463,7 +463,19 @@ def setupOutputs(self):
                 self._opReaders.append(op5)
 
             # Connect our external outputs to the internal operators we chose
-            self.Image.connect(providerSlot)
+            try:
+                self.Image.connect(providerSlot)
+            except DatasetConstraintError as initial_exception:  # might be thrown by opFeatureSelection
+                try:
+                    workflow = self.parent.parent.parent.parent
```

Comment:
okay this one now is very ugly.

Maybe it can be handled in `DataSelectionGui` in `handleDatasetConstraintError`. Subclassing `DataConstraintError` in `opFeatureSelection` and handling this accordingly through `DataSelectionGui` might be an option

Reply:
This is a bit tricky. 
The code should not be interrupted here, as there is nothing wrong with the selected dataset (except that the previously selected featureMatrix/Sigmas do not fit anymore => Subclassing...
But if we sublass and handle the error in DataSelectionGui, we still leave OpDataSelection.setupOutputs too early. Or we call setupOutputs again from within the 'handling'.
Maybe it it is the best option to go back to using the featureMatrix InputSlot in a kind of transaction slot.
Let's discuss this tomorrow in more detail.

Reply:
aye, I see. Good point :)